pkg/network/proxyimpl/hybrid/proxy.go

```diff
@@ -153,9 +158,21 @@ func (p *HybridProxier) OnServiceDelete(service *corev1.Service) {
 	p.switchedToUserspaceLock.Lock()
 	defer p.switchedToUserspaceLock.Unlock()
 
-	if isUsingUserspace, ok := p.usingUserspace[svcName]; ok && isUsingUserspace {
+	isUsingUserspace, ok := p.usingUserspace[svcName]
+	_, isPendingDeletion := p.pendingDeletion[svcName]
+
+	if (ok && isUsingUserspace) || isPendingDeletion {
 		klog.V(6).Infof("hybrid proxy: del svc %s/%s in unidling proxy", service.Namespace, service.Name)
 		p.unidlingProxy.OnServiceDelete(service)
+		if isPendingDeletion {
```

Comment:
I guess I could chalk this out to a separate function since the code is duplicated here and below.